common/src/main/java/com/revenuecat/purchases/common/networking/RCHTTPStatusCodes.kt

```diff
@@ -0,0 +1,10 @@
+package com.revenuecat.purchases.common.networking
+
+object RCHTTPStatusCodes {
+    const val SUCCESS = 200
+    const val CREATED = 201
+    const val UNSUCCESSFUL = 300
+    const val NOT_MODIFIED = 304
+    const val NOT_FOUND = 404
+    const val ERROR = 500
```

Comment:
kinda surprised that this isn't an enum, is there an advantage to having it be an object with constants instead?

Reply:
I did it with an enum initially like:

```
enum class RCHTTPStatusCodes(val code: Int) {
    SUCCESS(200),
    CREATED(201),
    UNSUCCESSFUL(300),
    NOT_MODIFIED(304),
    NOT_FOUND(404),
    ERROR(500),
}
```

To access the code, it needs to be done like:

```
result.responseCode < RCHTTPStatusCodes.ERROR.code,
```

Which is more verbose than just doing:

```
result.responseCode < RCHTTPStatusCodes.ERROR,
```

Apart from that some marginal benefit is that with const val, the value gets replaced automatically when the compiled code compiles to Java:

```
               boolean created = result.getResponseCode() == 201;
```
```
    LINENUMBER 359 L3
    ALOAD 1
    INVOKEVIRTUAL com/revenuecat/purchases/common/networking/HTTPResult.getResponseCode ()I
    SIPUSH 201
    IF_ICMPNE L4
    ICONST_1
    GOTO L5
   L4
    ICONST_0
   L5
    ISTORE 2
   L6
```

Whereas with enums, it would look like this:
```
result.getResponseCode() == RCHTTPStatusCodes.CREATED.getCode()
```
```
    LINENUMBER 200 L47
    ALOAD 1
    INVOKEVIRTUAL com/revenuecat/purchases/common/networking/HTTPResult.getResponseCode ()I
    GETSTATIC com/revenuecat/purchases/common/networking/RCHTTPStatusCodes.ERROR : Lcom/revenuecat/purchases/common/networking/RCHTTPStatusCodes;
    INVOKEVIRTUAL com/revenuecat/purchases/common/networking/RCHTTPStatusCodes.getCode ()I
    IF_ICMPGE L48
    ICONST_1
    GOTO L49
   L48
    ICONST_0
   L49
    INVOKESTATIC java/lang/Boolean.valueOf (Z)Ljava/lang/Boolean;
   L50
```

So this way, it's less overhead, but enums have the benefit of being able to be used as a type though, so it would be easier to use in a `when` statement for example.

What do you think?

Reply:
happy to go with either. Coming from Swift world, the enum verbosity looks very similar to what you do with Swift enums (it'd be `RCHTTPStatusCodes.ERROR.rawValue`), which is long but I guess I'm used to it 😂